cms/cache/permissions.py

```diff
@@ -55,8 +56,9 @@ def clear_user_permission_cache(user):
     Cleans permission cache for given user.
     """
     from django.core.cache import cache
-    for key in PERMISSION_KEYS:
-        cache.delete(get_cache_key(user, key), version=get_cache_permission_version())
+    for site in Site.objects.all():
```

Comment:
This could be a bit more efficient. Rather than grabbing each `Site` object, you could just grab their IDs because that's all that's actually getting used.

It'd become something like;

```python
def clear_user_permission_cache(user):
    """
    Cleans permission cache for given user.
    """
    from django.core.cache import cache
    for site_id in Site.objects.values_list('id', flat=True):
        for key in PERMISSION_KEYS:
            cache.delete(
                get_cache_key(user, "%s:%s" % (site_id, key)),
                version=get_cache_permission_version()
            )
```